### PR TITLE
Bugs/5953 bug bbox

### PR DESCRIPTION
**Pull request by utas-raymondng**
Fix bug where any input value from frontend will be handle correctly by the CQL BBOX by adjust it within [-180, 180], geo_bounding_box in elastic require bbox to work within this range

### CODE REVIEW

File: server/src/main/java/au/org/aodn/ogcapi/server/core/util/BboxUtils.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    public static Geometry normalizeBbox(double minx, double maxx, double miny, double maxy) {
+        // Bounding check, if greater than 360 already cover whole world, so adjust the maxx to something
+        // meaningful, noted that minx and maxx is not normalized yet so can be anything even beyond 180
+        if((maxx - minx) >= 360) {
+            // Value does not matter, it covered whole world which is equal to
+            minx = -180;
+            maxx = 180;
+        }
+        minx = (minx < -180) ? 180 - Math.abs(180 + minx) : minx;
+        maxx = (maxx > 180) ? -(maxx - 180) : maxx;
```
Comment (vietnguyengit) [inline server/src/main/java/au/org/aodn/ogcapi/server/core/util/BboxUtils.java:32]: there is a math formula for this: https://www.movable-type.co.uk/scripts/latlong.html, so we can avoid `?` comparing logics

> 	The longitude can be normalised to −180…+180 using (lon+540)%360-180

File: server/src/main/java/au/org/aodn/ogcapi/server/core/util/BboxUtils.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    protected static Geometry createBoxPolygon(double minx, double maxx, double miny, double maxy) {
+        // If the longitude range crosses the anti-meridian (e.g., maxx > 180)
+        // Normal case have not cross dateline
+        Coordinate[] coordinates = new Coordinate[] {
+                new Coordinate(minx, miny),   // Bottom-left corner
+                new Coordinate(maxx, miny),   // Bottom-right corner
+                new Coordinate(maxx, maxy),   // Top-right corner
+                new Coordinate(minx, maxy),   // Top-left corner
+                new Coordinate(minx, miny)    // Closing the loop (bottom-left corner)
+        };
```
[the remark this answers is not in the document]
Comment (utas-raymondng) [reply server/src/main/java/au/org/aodn/ogcapi/server/core/util/BboxUtils.java:56]: minx can > maxx may be it is better to call it startx and endx, same for miny can > maxy, it is just just mean it roate

File: server/src/main/java/au/org/aodn/ogcapi/server/core/util/BboxUtils.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        // Normalized the box, so it is within [-180, 180]
+        List<Polygon> polygons = new ArrayList<>();
+        if(maxx >= 0 && maxx <= 180) {
+            // Normal case
+            polygons.add((Polygon)createBoxPolygon(minx, maxx, miny, maxy));
+        }
+        else {
+            polygons.add((Polygon)createBoxPolygon(minx, 180, miny, maxy));
+            polygons.add((Polygon)createBoxPolygon(-180, maxx, miny, maxy));
+        }
```
[the remark this answers is not in the document]
Comment (utas-raymondng) [reply server/src/main/java/au/org/aodn/ogcapi/server/core/util/BboxUtils.java:43]: Can

File: server/src/main/java/au/org/aodn/ogcapi/server/core/util/BboxUtils.java (hunk 1 above)
Comment (utas-raymondng) [reply server/src/main/java/au/org/aodn/ogcapi/server/core/util/BboxUtils.java:32]: I tried to use this formula but the test will failed with the number off slightly

File: server/src/main/java/au/org/aodn/ogcapi/server/core/util/BboxUtils.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (utas-raymondng) [reply server/src/main/java/au/org/aodn/ogcapi/server/core/util/BboxUtils.java]: there are case where minx > maxx, that is your cross the line from say -180 to 170

**Review by vietnguyengit — APPROVED** [review]

**Pull request merged**